core/test/scripts/VoteGasEstimation.js

```diff
@@ -0,0 +1,137 @@
+const { RegistryRolesEnum } = require("../../utils/Enums.js");
+const { getRandomSignedInt, getRandomUnsignedInt } = require("../../utils/Random.js");
+
+const Registry = artifacts.require("Registry");
+const Voting = artifacts.require("Voting");
+const VotingToken = artifacts.require("VotingToken");
+
+const COMMIT_PHASE = "0";
+const secondsPerDay = web3.utils.toBN(86400);
+
+const moveToNextRound = async voting => {
+  const phase = await voting.getVotePhase();
+  const currentTime = await voting.getCurrentTime();
+  let timeIncrement;
+  if (phase.toString() === COMMIT_PHASE) {
+    timeIncrement = secondsPerDay.muln(2);
+  } else {
+    timeIncrement = secondsPerDay;
+  }
+
+  await voting.setCurrentTime(currentTime.add(timeIncrement));
+};
+
+const moveToNextPhase = async voting => {
+  const currentTime = await voting.getCurrentTime();
+  await voting.setCurrentTime(currentTime.add(secondsPerDay));
+};
+
+const getVoter = async (accounts, id) => {
+  // Offset by 2, because owner == accounts[0] and registeredDerivative == accounts[1]
+  return accounts[id + 2];
+};
+
+async function run() {
+  const voting = await Voting.deployed();
+  const votingToken = await VotingToken.deployed();
+  const registry = await Registry.deployed();
+
+  const accounts = await web3.eth.getAccounts();
+  const owner = accounts[0];
+  const registeredDerivative = accounts[1];
+
+  await voting.setInflationRate({ value: web3.utils.toWei("0.01", "ether") });
+  await registry.addMember(RegistryRolesEnum.DERIVATIVE_CREATOR, owner);
+
+  if (!(await registry.isDerivativeRegistered(registeredDerivative))) {
```

Comment:
We should really remove the registry's check against re-registering. In fact, the registry could really be simplified a lot considering how we currently use it.